topostats/tracing/dnatracing.py

```diff
@@ -127,94 +141,185 @@ def __init__(
             Degree of the spline.
         """
         self.image = image * 1e-9 if convert_nm_to_m else image
-        self.grain = grain
+        self.mask = mask
         self.filename = filename
         self.pixel_to_nm_scaling = pixel_to_nm_scaling * 1e-9 if convert_nm_to_m else pixel_to_nm_scaling
         self.min_skeleton_size = min_skeleton_size
-        self.skeletonisation_method = skeletonisation_method
+        self.mask_smoothing_params = mask_smoothing_params
+        self.skeletonisation_params = skeletonisation_params
+        self.pruning_params = pruning_params
         self.n_grain = n_grain
+        self.joining_node_length = joining_node_length
         self.number_of_rows = self.image.shape[0]
         self.number_of_columns = self.image.shape[1]
         self.sigma = 0.7 / (self.pixel_to_nm_scaling * 1e9)
-
-        self.gauss_image = None
-        self.grain = grain
+        # Images
+        self.smoothed_mask = np.zeros_like(image)
+        self.skeleton = np.zeros_like(image)
+        self.pruned_skeleton = np.zeros_like(image)
+        self.node_image = np.zeros_like(image)
+        self.ordered_trace_img = np.zeros_like(image)
+        self.fitted_trace_img = np.zeros_like(image)
+        self.splined_trace_img = np.zeros_like(image)
+        self.visuals = np.zeros_like(image)
+        # Nodestats objects
+        self.node_dict = None
+        self.node_image_dict = None
+        # Metrics
+        self.num_crossings = None
+        self.avg_crossing_confidence = None
+        self.min_crossing_confidence = None
+        self.topology = [None]
+        self.topology2 = [None]
+        self.contour_lengths = []
+        self.end_to_end_distances = []
+        self.mol_is_circulars = []
```

Comment:
Picking this up as a single comment as its the cause of a bunch of tests failing.

We don't have plural circular, here it is an array of molecule**s** that are being indicated as to whether they are circular or not.

```suggestion
        self.mols_are_circular = []
```

Will need all subsequent references changing.